### PR TITLE
use DecorativeMeshFile from Geometry::Mesh in sync. with latest Simbody. 

### PR DESCRIPTION
**Pull request by aymanhab**
And also to be consistent with the GUI.

### CODE REVIEW

**Comment by aymanhab** [comment]
These are mainly changes to wrapping necessary to make the GUI and API consistent (in using DecorativeMeshFile, workaround SWIG not supporting the "using" typdef. Failure on Appveyor of testComponents(BallJoint) is unrelated.

**Comment by aymanhab** [comment]
@chrisdembia Please let me know when you have a chance to review.

File: OpenSim/Simulation/Model/Appearance.h [hunk 1]
```diff
@@ -133,9 +137,9 @@ class OSIMSIMULATION_API Appearance : public Object {
     OpenSim_DECLARE_PROPERTY(color, SimTK::Vec3,
         "The color, (red, green, blue), [0, 1], used to display the geometry. ");
 
-    OpenSim_DECLARE_PROPERTY(surface_properties, SurfaceProperties,
+    OpenSim_DECLARE_UNNAMED_PROPERTY(SurfaceProperties,
         "Visuals applied to surfaces associated with this Appearance.");
-    OpenSim_DECLARE_PROPERTY(curve_properties, CurveProperties,
+    OpenSim_DECLARE_UNNAMED_PROPERTY(CurveProperties,
```
Comment (chrisdembia) [inline OpenSim/Simulation/Model/Appearance.h]: Oh, a bonus! This change likely requires updating updateFromXML(); perhaps that can be a separate PR.

File: OpenSim/Simulation/Model/ModelVisualizer.cpp [hunk 2]
```diff
@@ -245,7 +245,7 @@ bool ModelVisualizer::
 findGeometryFile(const Model& aModel, 
                  const std::string&          geoFile,
                  bool&                       geoFileIsAbsolute,
-                 SimTK::Array_<std::string>& attempts) const
+                 SimTK::Array_<std::string>& attempts)
```
Comment (chrisdembia) [inline OpenSim/Simulation/Model/ModelVisualizer.cpp]: Logically, this method is const. The other option is to const_cast `attempts` whenever you use it. But I think this is fine for now. No change neessary.

**Comment by chrisdembia** [comment]
My review is done. This PR looks good to go, but at least one related issue should be opened I think.

File: OpenSim/Simulation/Model/ModelVisualizer.cpp (hunk 2 above)
Comment (chrisdembia) [inline OpenSim/Simulation/Model/ModelVisualizer.cpp]: OH sorry I get it...it's static now. And `attempts` is passed in. Sorry.

Comment (chrisdembia) [inline OpenSim/Simulation/Model/ModelVisualizer.cpp]: I like that the method is static.

Comment (aymanhab) [inline OpenSim/Simulation/Model/ModelVisualizer.cpp]: Thanks @chrisdembia Yes, now a static method so can't be const since there's no object under it.

**Comment by aymanhab** [comment]
Thanks @chrisdembia Happy to open issue for XML conversion though models I have worked just fine, ignoring the name.
